redfish-core/lib/assembly.hpp

```diff
@@ -776,6 +780,186 @@ inline void handleChassisAssemblyPatch(
+        [asyncResp, assemblyUriPropPath, assemblyParentObjPath,
+         assembledObjPath,
+         assembledUriVal](const boost::system::error_code& ec,
+                          const associationList& associations) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR(
+                "DBUS response error [{}  : {}] when tried to get the Associations from [{}] to fill Assembly id of the assembled object [{}]",
+                ec.value(), ec.message(), assemblyParentObjPath.str,
+                assembledObjPath.str);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        std::vector<std::string> assemblyAssoc;
+        for (const auto& association : associations)
+        {
+            if (std::get<0>(association) != "assembly")
+            {
+                continue;
+            }
+            assemblyAssoc.emplace_back(std::get<2>(association));
+        }
+
+        if (assemblyAssoc.empty())
+        {
+            BMCWEB_LOG_ERROR(
+                "No assembly associations in the [{}] to fill Assembly id of the assembled object [{}]",
+                assemblyParentObjPath.str, assembledObjPath.str);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        // Mak sure whether the retrieved assembly associations are
+        // implemented before finding the assembly id as per bmcweb
+        // Assembly design.
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, assemblyUriPropPath, assemblyParentObjPath,
+             assembledObjPath, assemblyAssoc, assembledUriVal](
+                const boost::system::error_code& ec1,
+                const dbus::utility::MapperGetSubTreeResponse& objects) {
+            if (ec1)
+            {
+                BMCWEB_LOG_ERROR(
+                    "DBUS response error [{} : {}] when tried to get the subtree to check assembled objects implementation of the [{}] to find assembled object id of the [{}] to fill in the URI property",
+                    ec1.value(), ec1.message(), assemblyParentObjPath.str,
+                    assembledObjPath.str);
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (objects.empty())
+            {
+                BMCWEB_LOG_ERROR(
+                    "No objects in the [{}] to check assembled objects implementation to fill the assembled object [{}] id in the URI property",
+                    assemblyParentObjPath.str, assembledObjPath.str);
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            std::vector<std::string> implAssemblyAssocs;
+            for (const auto& object : objects)
+            {
+                auto it = std::ranges::find(assemblyAssoc.begin(),
+                                            assemblyAssoc.end(), object.first);
```

Comment:
```
                auto it = std::ranges::find(assemblyAssoc, object.first);
```

Reply:
Extremely sorry for making you point to the same kind of issue. Thanks for your patience. I will take extra care to ensure the next set is better adhered to these findings